From akpm@linux-foundation.org Wed Nov 21 15:07:41 2007
From: Lennert Buytenhek <buytenh@wantstofly.org>
Date: Wed, 21 Nov 2007 15:07:07 -0800
Subject: PCI: get rid of pci_dev::{vendor,device}_compatible fields
To: greg@kroah.com
Cc: akpm@linux-foundation.org, buytenh@wantstofly.org
Message-ID: <200711212307.lALN77ZP013894@imap1.linux-foundation.org>


From: Lennert Buytenhek <buytenh@wantstofly.org>

The vendor_compatible and device_compatible fields in struct pci_dev aren't
used anywhere, and are somewhat pointless.  Assuming that these are
historical artifacts, remove them.

Signed-off-by: Lennert Buytenhek <buytenh@wantstofly.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/pci.h |    5 -----
 1 file changed, 5 deletions(-)

--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -66,7 +66,6 @@ enum pci_mmap_state {
 #define PCI_DMA_FROMDEVICE	2
 #define PCI_DMA_NONE		3
 
-#define DEVICE_COUNT_COMPATIBLE	4
 #define DEVICE_COUNT_RESOURCE	12
 
 typedef int __bitwise pci_power_t;
@@ -167,10 +166,6 @@ struct pci_dev {
 	pci_channel_state_t error_state;	/* current connectivity state */
 	struct	device	dev;		/* Generic device interface */
 
-	/* device is compatible with these IDs */
-	unsigned short vendor_compatible[DEVICE_COUNT_COMPATIBLE];
-	unsigned short device_compatible[DEVICE_COUNT_COMPATIBLE];
-
 	int		cfg_size;	/* Size of configuration space */
 
 	/*
